src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5360,7 +5360,12 @@ private void BindMemberAccessReportError(BoundMethodGroup node, DiagnosticBag di
             }
             else
             {
-                if (boundLeft.Kind == BoundKind.TypeExpression ||
+
+                if ((object)boundLeft.Type == null)
```

Comment:
We had a bug here, that could be hit even without this change.
Typeless expressions could cause crashes or serialization asserts during error reporting.